### DISCUSSION TITLE
Backup stats

### DISCUSSION THREAD

**Discussion by Fish31520** [Nachtzuster/BirdNET-Pi#159] [Q&A]
Hi folks,

I was wondering if there's a way to backup all the stats of my Birdnet-Pi station.

One of my RPi recently fried and I was sad to loose all the informations gathered.

I don't want to keep all the recordings but a good database backup could be usefull to science and knowledge in the future.

So, is there a way to easily do that ?

Thanks a lot !

Jean

**Comment by alexbelgium** [comment]
Hi, this is being worked on by Nachtzuster here https://github.com/Nachtzuster/BirdNET-Pi/pull/119

If you don't want to wait for the feature, you could backup files by creating a cron job that would save every couple days your db and txt files on a USB drive. Here is an example code you can use for your cron script, assuming that your external drive is mounted to /backup :

Watch-out that "Recordings" would still be empty as it only shows when audio files exists
```
#!/bin/bash
current_date=$(date +%Y-%m-%d)
# Create a new directory with the current date under /backup
mkdir -p "/backup/$current_date"
# Find and copy all files ending by .conf ; .txt and .db
find "$HOME/BirdNET-Pi" -type f \( -name "*.conf" -o -name "*.db" -o -name "*.txt" \) -exec cp {} "/backup/$current_date" \;
echo "Files copied successfully to /backup/$current_date"
```

if you want to backup files you should backup "$HOME/BirdSongs/Extracted/Charts" and "$HOME/BirdSongs/Extracted/By_Date"

**Reply by Fish31520** [reply]
@alexbelgium 

Neet ! Thanks a lot !

Do you think datas could be sent on a Google (or others) Drive ?

**Reply by alexbelgium** [reply]
That would be very complex you can ask in the backup pr but don't have too much hopes

**Reply by Fish31520** [reply]
@alexbelgium

I was just curious, that's all ! 😅

Thanks a lot !

**Comment by Fish31520** [comment]
@Nachtzuster @alexbelgium

Hi,

I'm about to try the `backup_data.sh` script but I don't want to mess everything up ! 😅

First, if I get it right, from my PC, i've got to ssh to my birdnet RPi. It should look like :
`fish@fish:~$ ssh -X Birdnet@192.168.x.x`

Then I get `birdnetpi:~/BirdNET-Pi$ `
and I use the command `./scripts/backup_data.sh -a backup -f `
but after the `-f` argument, can I give the location of a local folder on my PC ?

Thanks for your help !

Jean

**Reply by alexbelgium** [reply]
Hi, you'll probably need to use scp to download the file after having run the backup script. For example if you generate your backup in /home/Birdnet/backupfile, you could download it in `scp -F /dev/null Birdnet@192.168.x.x:/home/Birdnet/backupfile /share/` and it will be available in your local `/share/backupfile` folder

I haven't tried yet the script though ;-)

**Reply by Nachtzuster** [reply]
Provided you have enough space on your RPi, you could definitely do it like that.
However in the Readme I'm recommending writing the backup to a separate medium mounted on the RPi (ie a thumbdrive or maybe a networkdrive)
That should be faster than the two step approach of creating+copying.

For a small dataset all this won't really matter

**Reply by Fish31520** [reply]
Hi,

What do you mean by "networkdrive" ? If my PC and my RPi are on the same network (they're both connected to the same wifi hotspot), does it means that the backup could be directly written on my PC without flooding my RPi memory ?

Thanks to you both for your help and your patience !

**Reply by alexbelgium** [reply]
If you have a SMB drive, you could mount it.

For example if you have a SMB at 192.1.1.1 with an exposed share "backup" with user "user" and password "password", you could you do :
```bash
# Main variables
disk="//192.1.1.1/backup"
USERNAME="user"
PASSWORD="password"

# Optional variables (uncomment if needed)
# CHARSET=",iocharset=utf8"
# SMBVERS=",vers=3.0" # Only define this if you need a specific SMB version
# SECVERS=",sec=ntlmv2" # Only enable if you need a specific SECVERS
# PUID=",uid=1000" # if your SMB supports it
# PGID=",gid=1000" # if your SMB supports it
# DOMAIN=",domain=workgroup"

# Building MOUNTOPTIONS
MOUNTOPTIONS="rw,file_mode=0775,dir_mode=0775,username=${USERNAME},password=${PASSWORD},nobrl"
MOUNTOPTIONS+="${SMBVERS:-}"
MOUNTOPTIONS+="${SECVERS:-}"
MOUNTOPTIONS+="${PUID:-}"
MOUNTOPTIONS+="${PGID:-}"
MOUNTOPTIONS+="${CHARSET:-}"
MOUNTOPTIONS+="${DOMAIN:-}"

# Mounting
sudo mkdir -p /mnt/SMB
sudo chown pi:pi /mnt/SMB
sudo mount -t cifs -o "$MOUNTOPTIONS" "$disk" /mnt/SMB
```

**Reply by Fish31520** [reply]
Hi,

Thanks for your help and your time.

Hum, the only thing I have is my laptop and the external driveI sometimes plug on it. Your solution may be a bit complex for me and my limited knowledge ! 😅

I think I'll keep it simple with that solution : https://github.com/Nachtzuster/BirdNET-Pi/discussions/224#top

Thanks anyway !

**Reply by alexbelgium** [reply]
Right click on the folder you want to share and you can share it as a smb drive in windows :) then you just put the up, name and password in the script and it should work!

Otherwise indeed the best solution is whatever works for you ;)
